### PR TITLE
cli: add --home flag to customize "Go Home" button

### LINKED ISSUE #2328: Allow customizing Go Home button
- Web Browser: Brave
- Local OS: Windows
- Remote OS: Arch
- Remote Architecture: amd64
- `code-server --version`: 

I noticed in the newest update, there is this button
image

which links directly to `
[microsoft/vscode](https://github.com/microsoft/vscode)

Is this a mistake? Official VSCode does not have this button

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR introduces a new cli flag `--home` which allows the user to customize the "Go Home" button seen in the application menu.

## Changes
- adds the "Go Home" button back to the menu (previously removed in vscode patch to prevent confusion: https://github.com/cdr/code-server/commit/5a38ab95fe0a0da3858254667335df7c1d2db57a)
- adds `--home` as a valid CLI flag to configure link for "Go Home" button

## Checklist
- [x] tested locally in development mode
- [x] updated the docs
- [x] updated a test

## Related
- #2245 

Fixes #2328

### CODE REVIEW

File: doc/FAQ.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - [Heartbeat File](#heartbeat-file)
 - [Healthz endpoint](#healthz-endpoint)
 - [How does the config file work?](#how-does-the-config-file-work)
+- [How do I customize the "Go Home" button?](#how-do-i-customize-the-go-home-button)
```
Comment (nhooyr) [inline doc/FAQ.md:25]: 👌🏼

**Review by nhooyr — COMMENTED** [review]
It appears my commit 5a38ab95fe0a0da3858254667335df7c1d2db57a was a lie and never actually disabled the home button?

Would be good to revert that in this PR then.

**Comment by nhooyr** [comment]
Commit separation is really on point @jsjoeio

File: ci/dev/vscode.patch [hunk 2]
```diff
@@ -4706,44 +4746,48 @@ index ff358cb6a10984868ed5a5aed5729ac6eb8ebeb7..c73be6d8e9f9b213aeee2b4c22b53fc5
...
    resolved "https://registry.yarnpkg.com/nice-try/-/nice-try-1.0.4.tgz#d93962f6c52f2c1558c0fbda6d512819f1efe1c4"
-@@ -6581,6 +6705,18 @@ optimist@^0.6.1:
+@@ -6581,29 +6705,29 @@ optimist@^0.6.1:
      minimist "~0.0.1"
      wordwrap "~0.0.2"
  
+-optionator@^0.8.2:
+-  version "0.8.2"
+-  resolved "https://registry.yarnpkg.com/optionator/-/optionator-0.8.2.tgz#364c5e409d3f4d6301d6c0b4c05bba50180aeb64"
+-  integrity sha1-NkxeQJ0/TWMB1sC0wFu6UBgK62Q=
```
[the remark this answers is not in the document]
Comment (jsjoeio) [reply ci/dev/vscode.patch:4756]: Not intentional. I can revert those.

**Comment by jsjoeio** [comment]
@nhooyr 
> it appears my commit 5a38ab9 was a lie and never actually disabled the home button?
> Would be good to revert that in this PR then.

No, it did disable the home button. I uncommented them but I'm having a hard time finding it in the commits. I can double-check, but I'm sure it's been reverted.

**Comment by jsjoeio** [comment]
If you go here, you can see where your changes (@nhooyr) were reverted: https://github.com/cdr/code-server/pull/2435/commits/0e7f504ebb902daad5a0f61440fddc89cc0190da#diff-9e295724ddcc33de1e1e16290c98d43f43e5ce542612dd015d982853be82493cL3869-L3872

File: ci/dev/vscode.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsjoeio) [reply ci/dev/vscode.patch:4756]: I think I reverted them but not 100% sure. Can you confirm @nhooyr ?

[the remark this answers is not in the document]
Comment (nhooyr) [reply ci/dev/vscode.patch:4756]: There's more now unfortunately. See the diff and expand the patch diff.

[the remark this answers is not in the document]
Comment (jsjoeio) [reply ci/dev/vscode.patch:4756]: Second attempt, did this work?

[the remark this answers is not in the document]
Comment (nhooyr) [reply ci/dev/vscode.patch:4756]: Nope, still there lol.

**Comment by jsjoeio** [comment]
Documenting: @nhooyr hopped on a Zoom call with me. For whatever reason, it worked when he did it and pushed his commit 🤷‍♂️

File: ci/dev/vscode.patch [hunk 3]
```diff
@@ -3859,16 +3902,13 @@ index 021af6e0f8983c492f9cdd048ba2dcae7640bc1d..4474a93beba03365709c3cda98b68213
...
+@@ -311,3 +311,6 @@ actionsRegistry.registerWorkbenchAction(SyncActionDescriptor.from(FocusPreviousP
  
  const workbenchRegistry = Registry.as<IWorkbenchContributionsRegistry>(WorkbenchExtensions.Workbench);
--workbenchRegistry.registerWorkbenchContribution(GoHomeContributor, LifecyclePhase.Ready);
-+// See https://github.com/cdr/code-server/issues/2328
-+// workbenchRegistry.registerWorkbenchContribution(GoHomeContributor, LifecyclePhase.Ready);
+ workbenchRegistry.registerWorkbenchContribution(GoHomeContributor, LifecyclePhase.Ready);
 +export const _1 = workbenchRegistry;
 +export const _2 = GoHomeContributor;
 +export const _3 = LifecyclePhase.Ready;
```
Comment (code-asher) [inline ci/dev/vscode.patch:3914]: Can these be removed as well? I don't think they're in VS Code's original source.

**Review by nhooyr — APPROVED** [review]
🚀

File: ci/dev/vscode.patch (hunk 3 above)
Comment (nhooyr) [reply ci/dev/vscode.patch:3914]: Ah that's true, these were part of my previous patch @jsjoeio

Comment (jsjoeio) [reply ci/dev/vscode.patch:3914]: Yeah, I'm not where those came from. Removing!

Comment (jsjoeio) [reply ci/dev/vscode.patch:3914]: I'm still having issues with the `yarn.lock` file. I will open a separate PR to address that!

**Comment by jsjoeio** [comment]
For anyone who comes here later, the reason I had to force-push was because I had to create a GPG key to sign and verify my commits. The only way to sign old commits was to rebase and force-push.

**Pull request merged**